### PR TITLE
Add Python 3.13 support using `legacy-cgi`

### PR DESCRIPTION
**Pull request by luhn**
A quick fix for #437 while #466 is in progress.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -53,6 +53,9 @@
     packages=find_packages("src", exclude=["tests"]),
     package_dir={"": "src"},
     python_requires=">=3.8",
+    install_requires=[
+        "legacy-cgi~=2.6; python_version >= '3.13'",
```
Comment (mmerickel) [inline setup.py:57]: I feel like `~=2.6` is too tight, would prefer to only pin a lower bound.

Comment (luhn) [reply setup.py]: Alright, done.

**Review by mmerickel — APPROVED** [review]
I'd love to get this shipped in a bugfix release as 1.8.9 if we can just to unblock python 3.13.

**Comment by luhn** [comment]
Should I rebase to 1.8 branch?

**Comment by luhn** [comment]
Looks like 1.8 branch didn't get 3.12 merged in, so we'd need to backport that as well.  https://github.com/Pylons/webob/pull/459

File: .github/workflows/ci-tests.yml [hunk 2]
```diff
@@ -21,35 +21,21 @@ jobs:
...
                   - py: "pypy-3.9"
                     toxenv: "pypy39"
                   - py: "pypy-3.10"
                     toxenv: "pypy310"
                 exclude:
-                    # Linux and macOS don't have x86 python
-                    - os: "ubuntu-latest"
-                      architecture: x86
-                    - os: "macos-latest"
-                      architecture: x86
```
Comment (digitalresistor) [inline .github/workflows/ci-tests.yml:46]: Has this been fixed?

**Comment by digitalresistor** [comment]
Line 44 in `.github/workflows/ci-tests.yml` needs to have the architecture removed:

https://github.com/Pylons/webob/blob/c3e3b5b06c2d6e7cab0bb4c64f85bec563ccf241/.github/workflows/ci-tests.yml#L44

File: .github/workflows/ci-tests.yml (hunk 2 above)
Comment (digitalresistor) [reply .github/workflows/ci-tests.yml:46]: No, it's all good, there are no errors!

Comment (digitalresistor) [reply .github/workflows/ci-tests.yml:46]: https://github.com/Pylons/webob/pull/467#issuecomment-2434174405

**Comment by luhn** [comment]
x86 support is already nuked by way of #468, so it shouldn't be in this PR.  I'm going to rebase to keep things tidy.

**Comment by luhn** [comment]
Done.

**Comment by digitalresistor** [comment]
Thanks @luhn

**Pull request merged**